src/modules/core/containers/pro-search-field/pro-search-field.tsx

```diff
@@ -0,0 +1,154 @@
+'use client';
+import { type FC, useMemo, useState } from 'react';
+
+import { TextField } from '@/modules/core/components/text-field';
+import { Button } from '@/modules/core/components/button';
+
+import type { ProSearchFieldProps } from './pro-search-field.interface';
+import { DropdownMenu } from '@/modules/core/components/dropdown-menu';
+import { useBoolean } from 'usehooks-ts';
+import { trpc } from '@/modules/core/utils/trpc.utils';
+import type { DropdownItem } from '@/modules/core/components/dropdown-menu/dropdown-menu.interface';
+import { getFullName } from '@/modules/user/utils/user.utils';
+import { useRouter } from 'next/navigation';
+import type { Prisma } from '@prisma/client';
+import { Avatar } from '@/modules/core/components/avatar';
+
+type ProDropdownItem = DropdownItem<
+  Prisma.ProfessionalGetPayload<{
+    include: { user: true };
+  }>
+>;
+
+const AutocompleteItemRender: FC<ProDropdownItem> = ({ data }) => {
+  if (!data) return null;
+
+  return (
+    <div className='flex items-center gap-x-2'>
+      <Avatar url={data.user.avatar} />
+      <span>{getFullName(data.user)}</span>
+    </div>
+  );
+};
+
+export const ProSearchField: FC<ProSearchFieldProps> = () => {
+  const router = useRouter();
+  // state
+  const isAutocompleteOpen = useBoolean();
+  const [searchQuery, setSearchQuery] = useState('');
+  // queries
+  const { data: professionalList, isFetching } =
+    trpc.professional.list.useQuery(
+      {
+        limit: 6,
+      },
+      {
+        enabled: !!searchQuery,
+      }
+    );
+  // memo
+  // @ts-ignore
```

Comment:
why ts-ignore?

Reply:
@IhorBalan There is some difference issue due to `companyId, createdAt, updatedAt`.

The difference between type:

```ts
type  ProfessionalItem = Prisma.ProfessionalGetPayload<{
    include: { user: true };
  }>
```

and response type of query

```
const { data } = trpc.professional.list.useQuery();

/* Types of property user are incompatible.
Type '{ professional: { id: string; userId: string; facebook: string | null; instagram: string | null; about: string null; createdAt: Date; updatedAt: Date; } | null; name: string | null; ... 11 more ...; bookings: { ...; }[]; }' is missing the following properties from type '{ id: string; avatar: string | null; firstName: string | null; lastName: string | null; phone: string | null; name: string | null; email: string | null; emailVerified: Date | null; ... 5 more ...; updatedAt: Date; }': companyId, createdAt, updatedAt */
const test: ProfessionalItem = data[0];  
```

Reply:
Please check it